Tests/DependencyInjection/Compiler/RemoveLoggingMiddlewarePassTest.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace Doctrine\Bundle\DoctrineBundle\Tests\DependencyInjection\Compiler;
+
+use Doctrine\Bundle\DoctrineBundle\DependencyInjection\Compiler\RemoveLoggingMiddlewarePass;
+use PHPUnit\Framework\TestCase;
+use Psr\Log\NullLogger;
+use Symfony\Component\Config\FileLocator;
+use Symfony\Component\DependencyInjection\ChildDefinition;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Loader\XmlFileLoader;
+
+final class RemoveLoggingMiddlewarePassTest extends TestCase
+{
+    public function testLoggingMiddlewareRemovedWhenLoggerMissing(): void
+    {
+        $container = $this->createContainer();
+        $container->compile();
+
+        $this->assertFalse($container->hasDefinition('logging_middleware_child'));
+    }
+
+    public function testLoggingMiddlewareNotRemovedWhenLoggerPresent(): void
+    {
+        $container = $this->createContainer();
+
+        $logger = (new Definition())
+            ->setClass(NullLogger::class);
+        $container->setDefinition('logger', $logger);
+
+        $container->compile();
+
+        $this->assertTrue($container->hasDefinition('logging_middleware_child'));
```

Comment:
Why is this to be an expected case? This child definition will still carry reference to non-existent logger service, won't it? Wouldn't that cause issues?

Reply:
This test is about the case where the logger is present 🙂